src/Bicep.Core/Emit/ExpressionConverter.cs

```diff
@@ -194,6 +194,11 @@ private LanguageExpression ConvertPropertyAccess(PropertyAccessSyntax propertyAc
                         return this
                             .CreateConverterForIndexReplacement(GetResourceNameSyntax(resourceSymbol), indexExpression, propertyAccess)
                             .ConvertExpression(GetResourceNameSyntax(resourceSymbol));
+                    case "location":
+                        // the location is dependent on the name expression which could involve locals in case of a resource collection
+                        return this
+                            .CreateConverterForIndexReplacement(GetResourceNameSyntax(resourceSymbol), indexExpression, propertyAccess)
+                            .ConvertExpression(resourceSymbol.UnsafeGetBodyPropertyValue(LanguageConstants.ResourceLocationPropertyName));
```

Comment:
This causes a bug. I'll fix it.

Reply:
What's the bug?

---
In reply to: [613687738](https://github.com/Azure/bicep/pull/2261#discussion_r613687738) [](ancestors = 613687738)

Reply:
Wait...the fix was overwrite. I must accidentally run force push somehow. Let me restore it. See the https://github.com/Azure/bicep/pull/2261/commits/5abe7345a025120e29f0a3d543d0991e36fcbe6f commit for the fix.

Reply:
It's weird that GitHub did not re-run merge validation.

Reply:
Nvm I force pushed the fix along with some newline normalizations...seems to be a GitHub UI bug that it's still showing old file content.